_LowLevelCode/cpp/cpp_communicator/serialiser/cpp_serialise.hpp

```diff
@@ -2,13 +2,14 @@
 
 #include <mex.h>
 
+#define DIM_MAX 4294967295
```

Comment:
what is this magic number? That's a very specific value

Reply:
You don't recognise 2^32-1? The max stored by a uint32?

Reply:
you're using 2^32-1 elsewhere in the code (`herbert_core/utilities/misc/hlp_serialise.m`). 

Does the language define that as a constant, so this can be clarified as  `#define DIM_MAX UINT32_MAX - 1`?

Reply:
If this line is to change, would using a `constexpr` here be better than a macro? Generally, having a real memory location for the number prevents some common pitfalls of macros.

Something like:
```suggestion
constexpr unsigned long DIM_MAX{4294967295};
```

Reply:
Can you add a comment that `// 2^32-1`